cluster/validate-cluster.sh

```diff
@@ -52,7 +52,12 @@ CLUSTER_READY_ADDITIONAL_TIME_SECONDS="${CLUSTER_READY_ADDITIONAL_TIME_SECONDS:-
 
 EXPECTED_NUM_NODES="${NUM_NODES}"
 if [[ "${REGISTER_MASTER_KUBELET:-}" == "true" ]]; then
-  EXPECTED_NUM_NODES=$((EXPECTED_NUM_NODES+1))
+  if [[ "${KUBE_EXPERIMENTAL_REPLICATE_EXISTING_MASTER:-}" == "true" ]]; then
+    NUM_MASTERS=$(get-master-replicas-count)
```

Comment:
The `get-master-replica-count` function will be called only if `KUBE_EXPERIMENTAL_REPLICATE_EXISTING_MASTER` is true. So, it will be called only on providers who support master replication. I guess we can make an assumption that all cloud providers which supports replication will deliver `get-master-replica-count` function.